subscription-sender/internal/queue/notification.go

```diff
@@ -0,0 +1,31 @@
+package queue
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/RinaDish/subscription-sender/internal/services"
+	"github.com/nats-io/nats.go"
+)
+
+func (queue *Queue) Subscribe(ctx context.Context) error {
+	_, err := queue.QueueConn.Subscribe(queue.subscriptionTopicName, func(msg *nats.Msg) {
+        queue.getMessage(msg, ctx)
+    })
+
+	if err != nil {
+        return err
+    }
+
+	queue.logger.Infof("Listening for messages on 'subscription' topic...")
+
+	return nil
+}
+
+func  (queue *Queue) getMessage(msg *nats.Msg, ctx context.Context) {
```

Comment:
This method parses message and notifies subscribers, but its name does not represent what it does at all

Reply:
done